source/VersionHandler/src/VersionHandler.cs

```diff
@@ -165,7 +166,7 @@ private class FilenameComponents
             if (!string.IsNullOrEmpty(version)) {
                 int dotIndex = version.IndexOf('.');
                 if (dotIndex > 0 && version.Length > dotIndex + 1) {
-                    if (!float.TryParse(version.Substring(0, dotIndex + 2), out result)) {
+                    if (!float.TryParse(version.Substring(0, dotIndex + 2), NumberStyles.Any, CultureInfo.InvariantCulture, out result)) {
```

Comment:
But Unity's version will always be in format "X.Y", which is not a valid float on German/Russian/etc. OSes. Therefore, `float.TryParse()` should ignore the locale.